shell/platform/windows/text_input_plugin.cc

```diff
@@ -44,7 +44,7 @@ void TextInputPlugin::CharHook(Win32FlutterWindow* window,
     return;
   }
   // TODO bug 30661
-  active_model_->AddCharacter(static_cast<char>(code_point));
+  active_model_->AddCharacter(code_point);
```

Comment:
As with the GLFW code, it's not clear to me that this is correct. The signature of CharHook here was just duplicated from the GLFW code; this seems like the right moment to fix it. The types in the chain from here back through to `Win32Window::MessageHandler` should be audited, given clear character types, and a conversion done if necessary at whatever boundary point makes sense (probably in `Win32Window::MessageHandler` itself?)

Reply:
It doesn't seem like my comment was fully addressed. This code is called from a case that includes both `WM_CHAR` and `WM_UNICHAR`. The former is UTF-16 according to the docs, and currently it's just being cast to a 32-bit value instead of converted.

Reply:
> It doesn't seem like my comment was fully addressed. This code is called from a case that includes both `WM_CHAR` and `WM_UNICHAR`. The former is UTF-16 according to the docs, and currently it's just being cast to a 32-bit value instead of converted.

I'll confirm the details you mentioned right away.

Reply:
You're right, I've added logic for dealing with special cases in UTF16, but please comment on where to put the code.